### PR TITLE
FIX Imported namespaces now correctly used to determine class inheritance

### LINKED ISSUE #3707: Namespaced PermissionProvider implementors are not detected
If you namespace an implementor of PermissionProvider, like so:

```php
<?php namespace SomeCompany\Permissions;

use PermissionProvider;

class CustomProvider implements PermissionProvider {
    public function providePermissions() {
        return array(
            'DESTROY_WORLD' => 'Destroy the world',
        );
    }
}
```

Changes won't be picked up, so the permission codes will not be shown when creating roles.

### PR DESCRIPTION
**Pull request by dhensby**
Fixes #3707

This fix allows the manifest builder to recognise Classes and Interfaces that have been used within namespaced classes or where namespaced classes have been imported with a `use` keyword.

Previously a classname or interface in the global scope explicitly needed to have a `\` prepended to the name in the `extends` or `implements` declarations for the manifest builder to recognise it.

Some examples of previously failing declarations that now pass are correctly recognised

```php
<?php
namespace SilverStripe\Test\Space;

use ModelAdmin as Admin;
use PermissionProvider, TestOnly as Test,HiddenClass;
use OtherInterface;
use \InterfaceName;

class MyExample extends Admin implements PermissionProvider, Test, HiddenClass, OtherInterface, InterfaceName {

    ...

}
```

This class would now (correctly) be picked up as a `ModelAdmin` subclass and an implementor of `PermissionProvider`;

Other declarations that were correctly interpreted remain so:

```php
<?php
namespace SilverStripe\Test\Space;

class MyClass extends \ModelAdmin {
    ...
}
```

### CODE REVIEW

**Comment by dhensby** [comment]
It broke all the things!

File: core/manifest/ClassLoader.php [hunk 1]
```diff
@@ -101,7 +101,7 @@ public function getItemPath($class) {
 	 * @return bool
 	 */
 	public function classExists($class) {
```
Comment (willmorgan) [inline core/manifest/ClassLoader.php]: Should this fn be renamed? (I know it would break semver but still)

Comment (dhensby) [inline core/manifest/ClassLoader.php]: It should be, but not as part of this bug fix

Comment (dhensby) [inline core/manifest/ClassLoader.php]: Ok, I moved this to it's own PR for better discussion #3904

File: core/manifest/ClassManifest.php [hunk 2]
```diff
@@ -333,6 +360,123 @@ public function handleDir($basename, $pathname, $depth) {
...
+					//normalised class name (PHP is case insensitive for symbols/namespaces
+					$lClass = strtolower($class);
+
+					//go through all the imports and see if the class exists within one of them
+					foreach ($imports as $alias => $import) {
+						//normalise import
+						$import = trim($import, '\\');
+
+						//if there is no string key, then there was no declared alias - we'll use the main declaration
+						if (is_int($alias)) {
```
Comment (tractorcow) [inline core/manifest/ClassManifest.php]: pyramid

Comment (tractorcow) [inline core/manifest/ClassManifest.php]: Maybe some more continue, break, or break out some of these inner loops into functions for readability?
